apps/web/src/views/BuyCrypto/hooks/useOnRampSignature.ts

```diff
@@ -52,12 +63,13 @@ export const useOnRampSignature = <selectData = GetOnRampSignatureReturnType>(
         walletAddress,
         externalTransactionId,
         onRampUnit,
+        theme,
       },
     ]),
     enabled: Boolean(externalTransactionId && quote && walletAddress && getIsNetworkEnabled(chainId)),
     queryFn: async ({ queryKey }) => {
       // eslint-disable-next-line @typescript-eslint/no-shadow
-      const { quote, externalTransactionId, chainId, onRampUnit } = queryKey[1]
+      const { quote, externalTransactionId, chainId, onRampUnit, theme } = queryKey[1]
```

Comment:
why do we read from queryKey instead of using these variables directly?

Reply:
no i alsways just have done it this way ever since i wroekd on mobile as we do this in the mobile useQuery hooks quite often

Reply:
is there any specific reason?

Reply:
no dnt think so but i would have though since were usingt he query key to detcect change, then extracting the vars from the query key you can be certain their the most recent variables, so as to prevent an unnessecary refetch